x-pack/legacy/plugins/alerting/server/alerts_client.ts

```diff
@@ -215,6 +216,34 @@ export class AlertsClient {
     return this.getAlertFromRaw(id, updatedObject.attributes, updatedObject.references);
   }
 
+  public async updateApiKey({ id }: { id: string }) {
+    const {
+      references,
+      attributes: { enabled },
+    } = await this.savedObjectsClient.get('alert', id);
+
+    if (enabled === false) {
+      throw Boom.badRequest(`Can't update API key on a disabled alert`);
+    }
+
+    const apiKey = await this.createAPIKey();
+    const username = await this.getUserName();
+    await this.savedObjectsClient.update(
```

Comment:
So wondering here if we also want to invalidate the API key that was there previously.  Seems like we're just going to leave behind a trail of unused but still valid API keys.

Reply:
YES, we should definitely do that.  Although, is there some chance an old one could be available - like cached on a different Kibana instance?  Seems possible.  Which would imply so check to see when using an api key if we get back a "key not found" response, and use that as a cache invaldation signal (re-read the key).

Reply:
There's a card for api key cleanup (https://github.com/elastic/kibana/projects/26#card-25354998). The problem that arises is we don't know if any scheduled actions rely on the old api key. To solve this we would need to change a few things and we were planning to do so under that card.